airflow/providers/amazon/aws/log/s3_task_handler.py

```diff
@@ -93,7 +94,10 @@ def close(self):
             # read log and remove old logs to get just the latest additions
             with open(local_loc) as logfile:
                 log = logfile.read()
-            self.s3_write(log, remote_loc)
+            success = self.s3_write(log, remote_loc)
+            keep_local = conf.getboolean('logging', 'KEEP_LOCAL_LOGS')
+            if success and not keep_local:
+                shutil.rmtree(os.path.dirname(local_loc))
```

Comment:
What if something new is written to `local_loc` between `s3_write()` and `shutil.rmtree()`?

Reply:
Is that possible? This happens on the close() of the handler. Is there any way another task handler has the same local_loc path and is trying to write to it?

Reply:
Yep, I think you are right. Given `local_loc` is decided by `ti` + `try_number`, plus it's in `close()`, my earlier question should not be valid.